lib/core/test/unit/Cardano/Wallet/Address/PoolSpec.hs

```diff
@@ -23,26 +25,26 @@ spec = do
             prop_consistent testPool
 
         it "generator satisfies prop_gap and prop_fresh" $
-            forAll genUsageWithGap $ \usage ->
-                let pool = fromUsage usage
+            forAll (genUsageForGap $ AddressPool.gap testPool) $ \usage ->
+                let pool = fromUsage testPool usage
                 in  all ($ pool) [prop_gap, prop_fresh]
 
-        it "sequence of updates preserves invariants"
-            prop_updates
+        it "sequence of updates preserves invariants" $
+            prop_updates testPool
```

Comment:
what about property like 
1. get to some pool state s1
2. clear pool -> no addresses here in the pool
3. update to state s1 -> state here the same as in point 1
etc? (or it is automatically tested somewhere)

Reply:
Hm, not sure. In order to test that `update` respects the internal invariants (`prop_gap`, `prop_fresh`), I figured that I could pick any reasonably complex sequence of updates (here `addrs1` and `addrs2`). If the code contained a bug, then it would show up in this sequence already; I'm not sure if a more systematic construction of update sequences would yield more information. 🤔 I also figured that apart from the internal invariants, the code is more or less a copy of `Data.Map`, and that testing that the `Map` is tracked correctly would be overkill. 😅

I would prefer to keep `removeN` for shrinking only, as it uses `loadUnsafe` and would have to be tested separately. In this way, we do not have to test this function: A buggy shrinker affects our ability to find the root cause of bugs, but it does not give rise to false positives or negatives.

Reply:
indeed, testing `Map` is overkill, but maybe having roundtrip : state -> clear -> state no. Up to you. was just thinking loudly

Reply:
> having roundtrip : state -> clear -> state 

You have convinced me — I have added a property `prop_updates_order` which creates a pool from a random sequence of updates and then compares this to a pool built from the `Used` addresses, but updated in the order of their indices. 🤓